### PR TITLE
[BUG] If allowance is unlocked beforehand, split button is disabled even with a valid form

### LINKED ISSUE #35: If  the transaction is running, we could change the text to `Splitting...` or something like that, so the user knows what is happening.
If  the transaction is running, we could change the text to `Splitting...` or something like that, so the user knows what is happening.

_Originally posted by @mariano-aguero in https://github.com/gnosis/conditional-tokens-explorer/pull/18#discussion_r454518060_

### LINKED ISSUE #111: [BUG] If allowance is unlocked beforehand, split button is disabled even with a valid form

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #111 
Closes #35

### CODE REVIEW

File: src/hooks/useAllowanceState.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const useAllowanceState = (allowanceMethods: AllowanceMethods, amount: Bi
 
   useEffect(() => {
     fetchAllowance()
-  }, [fetchAllowance])
+  }, [fetchAllowance, amount])
```
Comment (mariano-aguero) [inline src/hooks/useAllowanceState.tsx:47]: Mmm maybe the amount is don't needed

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr130--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr130--conditionaltokensexplorer.review.gnosisdev.com

**Comment by hpmaxi** [comment]
It works great!. There still an error (not related to allowance), if you input everything for a split but uses `wallet balance` the button doesn't get enabled, but if you use any number it does.

File: src/hooks/useAllowanceState.tsx (hunk 1 above)
Comment (mariano-aguero) [reply src/hooks/useAllowanceState.tsx]: Removed

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr130--conditionaltokensexplorer.review.gnosisdev.com

**Pull request merged**